### PR TITLE
Tune Makefile, allow running targets in parallel across modules

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -147,15 +146,13 @@ install-tools:
 run: otelcorecol
 	./bin/otelcorecol_$(GOOS)_$(GOARCH) --config ${RUN_CONFIG} ${RUN_ARGS}
 
-.PHONY: for-all
-for-all:
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply Makefile:151]: nope, that is the "for-all" rule in the contrib Makefile :)

[the remark this answers is not in the document]
Comment (dmitryax) [reply Makefile:151]: Oh, right :)

**Review by dmitryax — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -56,45 +56,44 @@ all-modules:
...
 .PHONY: gotest-with-cover
 gotest-with-cover:
-	@$(MAKE) for-all CMD="make test-with-cover"
+	@$(MAKE) for-all-target TARGET="test-with-cover"
 	$(GOCOVMERGE) $$(find . -name coverage.out) > coverage.txt
 
 .PHONY: goporto
 goporto:
-	@$(MAKE) for-all CMD="make porto"
+	porto -w --include-internal ./
```
Comment (codeboten) [inline Makefile:76]: nit: the target should be renamed to `porto` since it doesn't call go

File: Makefile [hunk 3]
```diff
@@ -147,15 +146,13 @@ install-tools:
...
-	  	echo "running $${CMD} in $${dir}" && \
-	 	$${CMD} ); \
-	done
+GOMODULES = $(ALL_MODULES) $(PWD)
+.PHONY: $(GOMODULES)
+MODULEDIRS = $(GOMODULES:%=for-all-target-%)
+for-all-target: $(MODULEDIRS)
+$(MODULEDIRS):
+	$(MAKE) -C $(@:for-all-target-%=%) $(TARGET)
+.PHONY: for-all-target
```
Comment (codeboten) [inline Makefile:155]:
```suggestion
.PHONY: for-all-target
for-all-target: $(MODULEDIRS)
$(MODULEDIRS):
	$(MAKE) -C $(@:for-all-target-%=%) $(TARGET)
```

**Review by codeboten — APPROVED** [review]
Couple of non-blocking comments.

File: Makefile (hunk 2 above)
Comment (bogdandrutu) [reply Makefile:76]: Haha :) I thought that we made a rule that if it is in the main `Makefile` we prefix with `go` and in the `Makefile.common` without :)

File: Makefile [hunk 4]
```diff
@@ -309,13 +306,13 @@ gensemconv:
...
 check-contrib:
 	@echo Setting contrib at $(CONTRIB_PATH) to use this core checkout
-	make -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector=$(CURDIR)"
-	make -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector/model=$(CURDIR)/model"
-	make -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod tidy -go=1.16"
-	make -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod tidy -go=1.17"
-	make -C $(CONTRIB_PATH) test
+	@$(MAKE) -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector=$(CURDIR)"
+	@$(MAKE) -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector/model=$(CURDIR)/model"
+	@$(MAKE) -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod tidy -go=1.16"
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply Makefile:311]: Done.

File: Makefile (hunk 3 above)
Comment (bogdandrutu) [reply Makefile:155]: Done.

**Pull request merged**